src/ScottPlot/Plottable/LegendItem.cs

```diff
@@ -23,5 +23,6 @@ public bool IsRectangle
             get { return lineWidth >= 10; }
             set { lineWidth = 10; }
         }
+        public bool plottableVisible;
```

Comment:
To be consistent with the rest of the API this should probably be named `IsVisible` and default to `true`

Reply:
Actually, legend items where this is `false` still appear in the legend, just shaded. 

Maybe a better name for this would be `IsHidden` or `IsDisabled` or `IsShaded` or something.

I guess I'm circling back around toward favoring your original, `IsPlottableVisible`

There's some nuance between "not visible so don't show in legend" vs "not visible but show in the legend anyway and make it gray"... maybe an `enum` would actually make sense here to better describe these 3 possible states?

```cs
enum LegendItemVisibility { NotShown, Shaded, Visible };
```

Reply:
Yes, just `isVisible` felt a bit to narrow. The thing is I added a modification of `UpdateLegendItems` so that it accounts for invisible objects. Maybe that nuance should be directly in the plottables, but I really did not want this very interactive-oriented feature to spill on the rest of ScottPlot.

Reply:
I solved this by requiring the `LegendItem` constructor to accept an `IPlottable`, then everyone who makes `LegendItem` passes-in `this` so legend items can "remember" the `Plottable` that created them (and call its `IsVisible` property)

With this change all plot types now supported in the pop-out legend 😎